core/templates/dev/head/pages/collection-player-page/collection-player-page.directive.ts

```diff
@@ -304,26 +301,35 @@ oppia.directive('collectionPlayerPage', ['UrlInterpolationService',
               var collectionAllowsGuestProgress = (
                 ctrl.whitelistedCollectionIdsForGuestProgress.indexOf(
                   ctrl.collectionId) !== -1);
-              if (!ctrl.isLoggedIn && collectionAllowsGuestProgress &&
-                  GuestCollectionProgressService.hasCompletedSomeExploration(
-                    ctrl.collectionId)) {
-                var completedExplorationIds = (
-                  GuestCollectionProgressService.getCompletedExplorationIds(
-                    ctrl.collection));
-                var nextExplorationId = (
-                  GuestCollectionProgressService.getNextExplorationId(
-                    ctrl.collection, completedExplorationIds));
-                ctrl.collectionPlaythrough = (
-                  CollectionPlaythroughObjectFactory.create(
-                    nextExplorationId, completedExplorationIds));
-              } else {
-                ctrl.collectionPlaythrough = (
-                  CollectionPlaythroughObjectFactory.createFromBackendObject(
-                    collectionBackendObject.playthrough_dict));
-              }
-
-              ctrl.nextExplorationId =
-                ctrl.collectionPlaythrough.getNextExplorationId();
+              UserService.getUserInfoAsync().then(function(userInfo) {
+                $rootScope.loadingMessage = '';
```

Comment:
> Looks reasonable to me, thanks @Jamesjay4199.
>
>Just to check, have you tested manually? We don't want another regression. Make sure that the page looks OK while the AJAX call is still loading (can you post screenshots)?

Yes @seanlip, I moved the stopped showing the loading message after the page has  user status has loaded to prevent any issues. So it shows loading untill all the necessary data is fetched.